### PR TITLE
Add foreach support to Span and ReadOnlySpan

### PR DESCRIPTION
**Pull request by stephentoub**
Depends on https://github.com/dotnet/coreclr/pull/14922

cc: @jkotas, @ahsonkhan, @KrzysztofCwalina, @VSadov

### CODE REVIEW

File: src/System.Memory/ref/System.Memory.cs [hunk 1]
```diff
@@ -72,6 +79,11 @@ namespace System
         public Span<T> Slice(int start, int length) { throw null; }
         public T[] ToArray() { throw null; }
         public bool TryCopyTo(Span<T> destination) { throw null; }
+        public ref struct Enumerator
+        {
+            public bool MoveNext() { throw null; }
+            public ref T Current { get { throw null; } }
```
Comment (ahsonkhan) [inline src/System.Memory/ref/System.Memory.cs]: What about adding Reset?
http://referencesource.microsoft.com/#mscorlib/system/collections/ienumerator.cs

Comment (stephentoub) [reply src/System.Memory/ref/System.Memory.cs]: Reset is legacy and isn't needed by foreach.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test Windows x86 Release Build please (https://github.com/dotnet/corefx/issues/25228)

**Pull request merged**